src/hrflow_connectors/core/connector.py

```diff
@@ -763,8 +765,22 @@ class ConnectorModel(BaseModel):
     description: str
     url: str
     type: ConnectorType
+    subtype: constr(regex=CONNECTOR_SUBTYPE_FORMAT_REGEX) = Field(
+        ..., description="Lowercased string with no spaces"
+    )
     actions: t.List[ConnectorAction]
 
+    @validator("subtype", pre=True, always=True)
+    def check_subtype(cls, value: str) -> str:
+        cleaned_value = value.lower()
+        if cleaned_value != value:
+            raise ValueError(f"ConnectorModel's `subtype` {value} must be lowercase.")
+        if " " in cleaned_value:
+            raise ValueError(
+                f"ConnectorModel's `subtype` {value} must not contain any spaces."
+            )
+        return cleaned_value
+
```

Comment:
J'ai lu ton commentaire qui explique que peut y avoir des exceptions `meta4` pour Cegid. 
Mon sentiment c'est que bon ca fait bcp de fois ou `subtype` c'est juste `name.lower().replace(" ", "")` ou eventuellement si tu veux être plus général et c'est pas forcement necessaire `re.sub(r"\s", "", name.lower())`

Donc pour éviter d'avoir a ecrire `subtype` dans la majorité des cas je ferais plutot comme ca 

```py
subtype: str = Field(regex=CONNECTOR_SUBTYPE_FORMAT_REGEX, description=...)

@validator("subtype", pre=True, always=True)
def check_subtype(cls, value: str | None, values: dict) -> str:
      if value is None:
            return values.get("name", "").lower().replace(" ", "")
      return value
```

Autrement si ca te va pas je pense que dans ta version actuelle tu peux juste laisser `subtype: str = Field(regex=CONNECTOR_SUBTYPE_FORMAT_REGEX, description=...)` sans validateur. A quoi sert le validateur il fois que tu as mis la regexp dans `Field` ou `constr` il s'occupe de faire ca lui même non ?

Reply:
@the-forest-tree 
L'idée c'est que le `subtype` soit défini par l'utilisateur et que celui-ci respecte le format attendu.
Si la validation se fait quand même sans le validator, je peux le retirer oui.

Reply:
@the-forest-tree 
J'avais aussi pensé a rendre le champ optionnel.
Dans ce cas, si le champ n'est pas renseigné, il serait déduit du champ `name` (lowercase + retrait d'espace).
Dans le cas `meta4`, il faudrait le renseigné car comportement différent.
Cela devrait permettre d'avoir un update qui ne soit pas **breaking change**.
Qu'en penses-tu ?